### ISSUE TITLE
Nested navigation question

### ISSUE THREAD

**Issue by felixSchl** [grahammendick/navigation#810]
Hey,

I am wondering how a set up like this should be possible:

```
tabs
  home
profile
  settings
    ...
```

Which as far as I understand would look like this on iOS:

```
tabs
  home
    profile
      settings
        ...
```

when I try set this up the transition from home -> profile glitches out the first time the transition happens but successive navigation between the scenes is fine.

https://github.com/user-attachments/assets/5b83bafd-49af-4965-82b4-30e22ea46ed0

While I have the chance I also wanted to just say thank you for this project. I was on the brink of insanity trying to get react-navigation/expo-router to do the most basic things. I am so happy this project exists. It's been awesome to use so far.

**Comment by grahammendick** [comment]
Hey, thanks for getting in touch and I’m really glad you’re enjoying the Navigation router 🙏 

You want nested tab navigation where there’s a stack within each tab, right? Have you seen [my twitter sample](https://github.com/grahammendick/navigation/blob/861313f890097f672a84c26aed9d5fbf8360c647/NavigationReactNative/sample/twitter/Tabs.js#L24)? It shows nested navigation on iOS. You create a primary `TabBar` and add a `NavigationStack`, with its own stack navigator, to each tab.
```jsx
<TabBar primary>
  <TabBarItem title="Home">
    <NavigationHandler stateNavigator={homeNavigator}>
      <NavigationStack>
        <Scene stateKey="home"><Home /></Scene>
        <Scene stateKey="profile"><Profile /></Scene>
        <Scene stateKey="settings"><Settings /></Scene>
      </NavigationStack>
    </NavigationHandler>
```
This nested stack also works on Android. But I purposely didn’t demo that in the twitter sample because it’s more common to have  a single stack on Android.

I hope that helps but if I’ve misunderstood or anything isn’t clear please let me know.

**Comment by grahammendick** [comment]
Or maybe I misunderstood and you've got nested tabs working but you want the 'profile' scene to be outside of the tabs? You can do this on iOS using the `hidesTabBar` prop of the `NavigationStack`.
```jsx
<TabBar primary>
  <TabBarItem title="Home">
    <NavigationHandler hidesTabBar={({key}) => key === 'profile'} stateNavigator={homeNavigator}>
      <NavigationStack>
        <Scene stateKey="home"><Home /></Scene>
        <Scene stateKey="profile"><Profile /></Scene>
        <Scene stateKey="settings"><Settings /></Scene>
      </NavigationStack>
    </NavigationHandler>
```

**Comment by felixSchl** [comment]
I've gotten that part working but what I couldn't figure out how to add another level of nesting. Perhaps it does not even make sense in native apps; I am only exploring my options. What I am trying to achieve is to have the settings scene in this example have access to the profile route params & maybe even a react context.

Perhaps a more practical example, say you had a list of weightlifting programs in your app. When the user presses on one of the programs, they see a "detail" screen. Then on that detail screen they might want to join the program so they navigate to the "onboarding" screen. When I implement the onboarding screen, how do I know which program we're working with? Maybe I am stuck in the web way of thinking about it as "/program/:id/settings" where the navigation encodes that context. Perhaps I should have some mutable state in my app that I write to when I load the "detail" screen using `useNavigated()`?

**Comment by grahammendick** [comment]
You can keep passing the program id between scenes, for example, `stateNavigator.navigate('onboarding’, {programId})`. But if you don’t like that or have lots of data you want to share then you can use a centralised store like React Context or a 3rd party state library for React. You can write to it when the user selects the program rather than `useNavigated` (but, it’s up to you).
```jsx
onPress={()=> {
  setProgramId(programId) // React Context or 3rd party state
  stateNavigator.navigate('program', {programId});
}}
```

**Comment by felixSchl** [comment]
Thank you for your thoughts and confirming said approach. Gives me some reassurance I am on the right path. Thanks again for this wonderful library.

**Comment by grahammendick** [comment]
Thanks. I'm interested in getting to the bottom of the glitch you showed in the video. Could you maybe share your navigation setup or upload the project to github if it's shareable, please?

**Comment by felixSchl** [comment]
I did this, essentially:

```
<NavigationStack>
  <Scene stateKey="tabs">
    <NavigationBar hidden />
    <TabBar primary bottomTabs>
      <TabBarItem title="Programs">
        <NavigationHandler stateNavigator={programsNavigator}>
          <NavigationStack>
            <Scene stateKey="programs">
              ...
            </Scene>
            <Scene stateKey="program">
              <NavigationHandler stateNavigator={programNavigator}>
              <NavigationStack>
                <Scene stateKey="program_index">
                  ...
                </Scene>
              </NavigationStack>
            </Scene>
          </NavigationStack>
        </NavigationHandler>
      </TabBarItem>
    </Tabbar>
  </Scene>
</NavigationStack>
```

**Comment by grahammendick** [comment]
You only need one `NavigationStack` inside the `TabBarItem`. You don't need the program one. 
The 'programs' list scene and the 'program' details scene should both be inside the same stack.

**Comment by felixSchl** [comment]
Yup. I was hoping to use the react hierarchy to set up context but after creating this issue and talking it through I think just having sibling scenes and passing nav params around makes more sense.

**Comment by grahammendick** [comment]
It's also important to keep just the one stack so you get the native iOS header/title animation as you navigate back and forward between screens
